File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/BackupApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.backup.BackupManager

class BackupApi internal constructor(private val manager: BackupManager) {
    suspend fun create(passphrase: CharArray): Result<ByteArray> = manager.createBackup(passphrase)
    suspend fun restore(backupData: ByteArray, passphrase: CharArray): Result<Int> =
        manager.restoreBackup(backupData, passphrase)
}

File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/BackupApi.swift
import Foundation

/// Facade for encrypted backup and restore operations.
public struct BackupApi {
    private let manager: BackupManager

    init(manager: BackupManager) {
        self.manager = manager
    }

    public func create(passphrase: String) async throws -> Data {
        try await manager.createBackup(passphrase: passphrase)
    }

    public func restore(backupData: Data, passphrase: String) async throws -> Int {
        try await manager.restoreBackup(backupData: backupData, passphrase: passphrase)
    }
}
